plugins/outputs/influxdb_v3/influxdb_v3.go

```diff
@@ -0,0 +1,220 @@
+//go:generate ../../../tools/readme_config_includer/generator
+package influxdb_v3
+
+import (
+	"context"
+	"crypto/tls"
+	_ "embed"
+	"errors"
+	"fmt"
+	"math/rand"
+	"net"
+	"net/url"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/common/ratelimiter"
+	commontls "github.com/influxdata/telegraf/plugins/common/tls"
+	"github.com/influxdata/telegraf/plugins/outputs"
+	"github.com/influxdata/telegraf/plugins/serializers/influx"
+)
+
+//go:embed sample.conf
+var sampleConfig string
+
+type InfluxDB struct {
+	URLs               []string          `toml:"urls"`
+	LocalAddr          string            `toml:"local_address"`
+	Token              config.Secret     `toml:"token"`
+	Database           string            `toml:"database"`
+	DatabaseTag        string            `toml:"database_tag"`
+	ExcludeDatabaseTag bool              `toml:"exclude_database_tag"`
+	Timeout            config.Duration   `toml:"timeout"`
+	HTTPHeaders        map[string]string `toml:"http_headers"`
+	HTTPProxy          string            `toml:"http_proxy"`
+	UserAgent          string            `toml:"user_agent"`
+	ContentEncoding    string            `toml:"content_encoding"`
+	UintSupport        bool              `toml:"influx_uint_support"`
+	OmitTimestamp      bool              `toml:"influx_omit_timestamp"`
+	PingTimeout        config.Duration   `toml:"ping_timeout"`
+	ReadIdleTimeout    config.Duration   `toml:"read_idle_timeout"`
+	Log                telegraf.Logger   `toml:"-"`
+	commontls.ClientConfig
+	ratelimiter.RateLimitConfig
+
+	clients    []*httpClient
+	encoder    internal.ContentEncoder
+	serializer ratelimiter.Serializer
+	tlsCfg     *tls.Config
+}
+
+func (*InfluxDB) SampleConfig() string {
+	return sampleConfig
+}
+
+func (i *InfluxDB) Init() error {
+	// Set defaults
+	if i.UserAgent == "" {
+		i.UserAgent = internal.ProductToken()
+	}
+
+	if len(i.URLs) == 0 {
+		i.URLs = append(i.URLs, "http://localhost:8181")
+	}
+
+	// Init encoding if configured
+	switch i.ContentEncoding {
+	case "", "gzip":
+		i.ContentEncoding = "gzip"
+		enc, err := internal.NewGzipEncoder()
+		if err != nil {
+			return fmt.Errorf("setting up gzip encoder failed: %w", err)
+		}
+		i.encoder = enc
+	case "identity":
+	default:
+		return fmt.Errorf("invalid content encoding %q", i.ContentEncoding)
+	}
+
+	// Setup the limited serializer
+	serializer := &influx.Serializer{
+		UintSupport:   i.UintSupport,
```

Comment:
Why not always enable? It is supported by InfluxDBv3.

Reply:
I have to clarify what the current implementation status is. Thanks for the heads-up! If uint is supported I think we should unconditionally enable it.

Reply:
https://docs.influxdata.com/influxdb3/cloud-serverless/reference/sql/data-types/#unsigned-integers